external/concealer/include/concealer/utility/service_with_validation.h

```diff
@@ -0,0 +1,97 @@
+// Copyright 2015 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CONCEALER__SERVICE_WITH_VALIDATION_HPP_
+#define CONCEALER__SERVICE_WITH_VALIDATION_HPP_
+
+#include <chrono>
+#include <string>
+
+#include "concealer/autoware.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+namespace concealer
+{
+template <class T>
+class ServiceWithValidation
+{
+public:
+  ServiceWithValidation(std::string serviceName, Autoware & autoware)
+  : serviceName(serviceName),
+    logger(autoware.get_logger()),
+    client(autoware.create_client<T>(serviceName, rmw_qos_profile_default)),
+    validation_rate(std::chrono::seconds(1))
+  {
+  }
+
+  void operator()(const typename T::Request::SharedPtr & request)
+  {
+    validateAvailability();
+    callWithTimeoutValidation(request);
+  }
+
+  void operator()(const typename T::Request::SharedPtr & request, size_t attempts_count)
+  {
+    validateAvailability();
+    for (size_t attempt = 0; attempt < attempts_count; ++attempt, validation_rate.sleep()) {
+      std::optional<typename rclcpp::Client<T>::SharedFuture> future_opt =
```

Comment:
Do not use abbreviations such as `opt` unless your system imposes a limit such as a maximum length of characters.Also, the information given by a variable name of `future_opt` is not useful, since it is exactly the same information as the type.
A variable name should be a description of its use or role, not a type name, unless it is truly generic.

Reply:
Also, change the variable `future_opt` so that it is not a local variable, but only scoped within an if statement (use the "if statements with initializer" syntax).